src/main/java/com/esri/samples/na/closest_facility_static/ClosestFacilityStaticSample.java

```diff
@@ -0,0 +1,344 @@
+                  incidentsList.add(new Incident(incidentFeature.getGeometry().getExtent().getCenter()));
+                }
+
+              } catch (InterruptedException | ExecutionException e) {
+                displayMessage("Error retrieving list of incidents", e.getMessage());
+              }
+            });
+
+            // enable button and resolve button press
+            solveRoutesButton.setDisable(false);
+            solveRoutesButton.setOnAction(e -> {
+              try {
+                solveRoutes();
+              } catch (Exception ex) {
+                ex.printStackTrace();
+              }
+            });
+
+            // handle reset button press
+            resetButton.setOnAction(e -> {
+              try {
+                resetRoutes();
+              } catch (Exception ex) {
+                ex.printStackTrace();
+              }
+            });
+          }
+        });
+      });
+
+      // add the map view and control panel to stack pane
+      stackPane.getChildren().addAll(mapView, controlsVBox);
+      StackPane.setAlignment(controlsVBox, Pos.TOP_LEFT);
+      StackPane.setMargin(controlsVBox, new Insets(10, 0, 0, 10));
+
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+  }
+
+  // task to find the closes route between an incident and a facility
+  private void solveRoutes() {
+
+    // start the routing task
+    closestFacilityTask.loadAsync();
+
+    try {
+      closestFacilityTask.addDoneLoadingListener(() -> {
+        if (closestFacilityTask.getLoadStatus() == LoadStatus.LOADED) {
+          try {
+            // create default parameters for the task and add facilities and incidents to parameters
+            ClosestFacilityParameters closestFacilityParameters = closestFacilityTask.createDefaultParametersAsync().get();
+            closestFacilityParameters.setFacilities(facilitiesList);
+            closestFacilityParameters.setIncidents(incidentsList);
+
+            // solve closest facilities
+            try {
+              // use the task to solve for the closest facility
+              ListenableFuture<ClosestFacilityResult> closestFacilityTaskResult = closestFacilityTask.solveClosestFacilityAsync(closestFacilityParameters);
+              closestFacilityTaskResult.addDoneListener(() -> {
+                try {
+                  ClosestFacilityResult closestFacilityResult = closestFacilityTaskResult.get();
+
+                  // find the closest facility for each incident
+                  for (int indexOfIncident = 0; indexOfIncident < incidentsList.size(); indexOfIncident++) {
+
+                    // get the index of the closest facility to incident. (i) is the index of the incident, [0] is the index of the closest facility.
+                    Integer closestFacilityIndex = closestFacilityResult.getRankedFacilityIndexes(indexOfIncident).get(0);
+
+                    // get the route to the closest facility.
```

Comment:
be consistent with comment style throughout, usually no full stop at the end of these comments